src/mewbot/api/registry.py

```diff
@@ -121,14 +165,48 @@ def do_register(api: Type[Component]) -> Type[Component]:
 
     @classmethod
     def _detect_api_versions(mcs, impl: Type[Any]) -> Iterable[Tuple[ComponentKind, str]]:
+        """Finds all API versions that match a given implementation."""
+
         for kind, apis in mcs._api_versions.items():
             for version, api in apis.items():
                 if issubclass(impl, api):
                     yield kind, version
 
     @classmethod
-    def api_version(mcs, component: Component) -> Tuple[ComponentKind, str]:
-        for val in mcs._detect_api_versions(type(component)):
+    def api_version(mcs, component: Component | Type[Component]) -> Tuple[ComponentKind, str]:
+        """
+        Gets the Component Kind (e.g. 'Behaviour') and API version (e.g. 'v1') from a component.
+
+        This function will only return the first matched type, as per the Registry's
+        'one base type' constraint.
+        """
+
+        if not isinstance(component, type):
+            component = type(component)
+
+        for val in mcs._detect_api_versions(component):
             return val
 
         raise ValueError(f"No API version for {component}")
+
+    @staticmethod
+    def load_and_register_modules(name: Optional[str] = None) -> Iterable[Any]:
+        """
+        Load modules from setuptools that declare a `mewbot-v1` entrypoint.
+
+        This looks at the entrypoints of all modules which are locatable with
+        importlib,
+
+        :param str name: if given, loads only plugins with the given `name`.
+        :return: the loaded objects.
+        """
+        for distribution in importlib_metadata.distributions():  # type: ignore
+            for entry_point in distribution.entry_points:
+                if entry_point.group not in API_DISTRIBUTIONS:
+                    continue
```

Comment:
On the thought of error dectection, is it worth saying "This item's not being picked up because of this reason"? so we're not like "Why is my module not loading? :("

Reply:
That message would occur for every python module installed in the current context, which I don't think would be productive for anyone?

Reply:
ah. that's fair yeah maybe let's not do that. Is there some other confirmation of what modules load?

Reply:
Resolved via direct discussion